packages/button/src/Button/Button.tsx

```diff
@@ -68,18 +68,32 @@ const ButtonContext = React.createContext<
   active: false,
 });
 
+type DeriveStateParams = {
+  state: Required<RootBaseProps>["state"];
+  disabled?: boolean;
+  active?: boolean;
+};
+
+// Derive the state of the button based on the html button props in
+const deriveState = (params: DeriveStateParams): InternalProps["state"] => {
+  if (params.disabled) return "disabled";
+  if (params.state === "loading") return "loading";
+  if (params.active) return "active";
+  return params.state;
+};
+
```

Comment:
Making `state` derived from props like `active` and `disabled` so that the button derives to a predictable state